src/store/actions/Chat.ts

```diff
@@ -0,0 +1,154 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import firebase from "firebase/app";
+
+export const SEND_GLOBAL_CHAT: string = "SEND_GLOBAL_CHAT";
+export const SEND_ROOM_CHAT: string = "SEND_ROOM_CHAT";
+export const SEND_PRIVATE_CHAT: string = "SEND_PRIVATE_CHAT";
+export const SEND_TABLE_CHAT: string = "SEND_TABLE_CHAT";
```

Comment:
`@reduxjs/toolkit` has action creators that avoid the need for this level of boilerplate, was there a specific reason you went this way?

https://redux-toolkit.js.org/tutorials/basic-tutorial#introducing-createaction